### PR TITLE
msg/simple/Pipe: handle addr decode error

### PR DESCRIPTION
**Pull request by liewegas**
Fixes: http://tracker.ceph.com/issues/18072
Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

**Comment by athanatos** [comment]
This looks right to me as far as it goes.  Is the supposition that we didn't get the whole buffer?

**Comment by liewegas** [comment]
I think it must have been corrupted on the wire or something.  In any case, the test crashed instead of just faulting the connection.

**Comment by athanatos** [comment]
Ok, lgtm

**Comment by gregsfortytwo** [comment]
lgtm too

File: src/msg/simple/Pipe.cc [hunk 1]
```diff
@@ -1043,9 +1043,15 @@ int Pipe::connect()
     goto fail;
   }
   {
-    bufferlist::iterator p = addrbl.begin();
-    ::decode(paddr, p);
-    ::decode(peer_addr_for_me, p);
+    try {
```
Comment (tchaikov) [inline src/msg/simple/Pipe.cc]: nit, could remove the outer braces.

File: src/msg/simple/Pipe.cc [hunk 2]
```diff
@@ -1043,9 +1043,15 @@ int Pipe::connect()
...
+      bufferlist::iterator p = addrbl.begin();
+      ::decode(paddr, p);
+      ::decode(peer_addr_for_me, p);
+    }
+    catch (buffer::error& e) {
```
Comment (tchaikov) [inline src/msg/simple/Pipe.cc]: and remove `e` if it is not referenced in the `catch` block.

**Comment by liewegas** [comment]
updated.  thanks!

**Comment by liewegas** [comment]
retest this please

**Pull request merged**